### PR TITLE
2016-03-08-best-practices-for-appdynamics-business-transaction-identification

### PR DESCRIPTION
**Pull request by mattsonjoshcf**
**    author(s) mini bio, soft capped to 75 words**
Senior Application Support Engineer and 6 year Racker on the Critical Application Services team.  Specializing in Java web applications
**    author(s) social media contact info**
N/A
**    accept or decline to allow redaction, editing or structure changes by technical writer staff**
Accept
**    desired date of publishing**
ASAP
**    whether or not you're willing to answer comments about your post (see step #5 above)**
No

### CODE REVIEW

**Comment by ktbartholomew** [comment]
@mattsonjoshcf The build is currently failing because you've added the category "Tools" and there are no other posts using that category. I'd recommend removing that category from this post.

@catlook Would you mind reviewing this post?

**Comment by mattsonjoshcf** [comment]
@ktbartholomew Thanks for the update!  I've removed the Tools category as suggested.

File: _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+comments: false
+author: Josh Mattson
+published: true
+authorIsRacker: true
+authorAvatar: 'http://www.gravatar.com/avatar/791617263c70278859e1b26c15d13eab'
+categories:
+    - DevOps
+---
+
+[AppDynamics](https://www.appdynamics.com/) is a powerful Application Performance Management tool that, properly configured, can provide tremendous insight in to application and infrastructure performance bottlenecks and enable operations and development teams to rapidly identify and resolve issues.  Though AppDynamics will begin working to collect and measure application performance data out of the box, some configuration and customization is necessary in order to reach its full capabilities.  This guide will explain and clarify best practices around how to identify your application's critical Business Transactions in order to get the most out of AppDynamics and ultimately, the most out of your application and infrastructure.
```
Comment (catlook) [inline _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md]: ... resolve issues. Although AppDynamics collects and measures application ...
.. guide explains and clarifies best practices ...  [ ed note: really I'd use explain or clarify, but i don't think you need both ]
...  critical business transactions in  ...  [ ed note: if want to  emphasize, use italics, unless Business Transaction is a very specialized term in this context ]
... AppDynamics and, ultimately, the ...

File: _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+<!-- more -->
+
+### Best Practices Derived from Years of Experience
+
+Through years of experience utilizing AppDynamics to support hundreds of varying  customer applications, the Rackspace [Critical Application Services](https://www.rackspace.com/en-us/enterprise-cloud-solutions/critical-applications) team of Application Support Engineers have developed a streamlined way to quickly identify, configure and start measuring the most important business transactions in any given application.
```
Comment (catlook) [inline _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md]: ... configure, and start ...

File: _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+### Business Transactions Conceptualized
+
+AppDynamics introduces several concepts that are important to grasp in order to ensure that you're getting the most relevant and actionable data including Business Applications, Tiers, Nodes and Business Transactions.  For this guide, we'll be focusing solely on Business Transactions.
```
Comment (catlook) [inline _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md]: ... including business applications, tiers, nodes and business transactions.  For ...
... solely on business transactions.  [ ed note: again...italics? ]

File: _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+In simple terms, a Business Transaction is the end to end process that is taken through your application to serve any given request.  When conceptualizing these transactions, it's best to think of them from a business or end-user perspective (e.g. What is the end-user doing when they execute this transaction?).
```
Comment (catlook) [inline _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md]: ... a business transaction is the end-to-end process ...

File: _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+* Home Page
+* Browse
+* Search
+* Category Page
+* Product Details
+* Add to Cart
+* Checkout
+
+Depending on the scope of the application, it may be necessary to involve other teams and stakeholders within your business to ensure that all critical paths through the application have a Business Transaction defined within AppDynamics.
```
Comment (catlook) [inline _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md]: ... have a business transaction defined ...

File: _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+![Sample Business Transactions in AppDynamics]({% asset_path 2016-03-08-best-practices-for-appdynamics-business-transaction-identification/appdynamics_business_transactions.png %})
```
Comment (catlook) [inline _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md]: Sample business transactions in AppDynamics ...

File: _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+As you can see in the image and examples provided above, when naming Business Transactions in AppDynamics, it's important to name them in a conversational way.  For example, we may be defining a business transaction based on the servlet's entry URI (e.g. /store/browse/cat), but this would typically be a poor name for a transaction.  Instead, think if you noticed an issue in the application at 2:00 AM and needed to call a teammate to look in to the issue.  Would you say 'We're having an issue with category pages not serving properly' or would you say 'We're having an issue with _slash store slash browse slash cat_'?
```
Comment (catlook) [inline _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md]: As you can see in the preceding image and examples, when naming business transactions in AppDynamics, it's ...

File: _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+As a final word of caution, we need to be cognizant of the fact that AppDynamics's primary purpose is to further development efforts and/or troubleshoot performance issues during load tests or P1 production issues.  Often, when implementing AppDynamics in a new application, we may see what appears to be 60-100+ transactions identified as critical.  It may be counter-intuitive at first, but it's important to try to limit the number of Business Transactions as having too many can cause confusion for users, excessive overhead on your application, can potentially cause AppDynamics to miss out on correlating information and generally muddy the waters if transactions are *too* granular.  Ideally, we should aim for no more than 10-20 critical business transactions per tier.  Of course, there are always exceptions to rules, but this is a good one to keep in the back of your mind as you go through this process.
```
Comment (catlook) [inline _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md]: ... efforts and to troubleshoot ...
... we might see ...
... of business transactions. Having too many transactions causes confusion for users, results in excessive overhead on your application, and generally muddies the waters if transactions are *too* granular. An overabundance of transactions might even cause AppDynamics to miss out on correlating information.  Ideally, ...

File: _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+Once this process is complete, we should have a solid list of the most important Business Transactions that need to be configured and profiled in order to provide us with the most appropriate and actionable data.
```
Comment (catlook) [inline _posts/2016-03-08-best-practices-for-appdynamics-business-transaction-identification.md]: ... important business transactions that ...
[ ed note: Thanks for sharing!!  Use any of my edits that resonate and stir fry the rest with a little olive oil and ginger! - Cat ]

**Comment by smashwilson** [comment]
Whoops, this slipped through the cracks for a while. Sorry about that! It's merged and shipping presently.

**Pull request merged**